charts/nr-k8s-otel-collector/tests/ksm_selector_test.yaml

```
@@ -0,0 +1,106 @@
+suite: ksmSelector
+templates:
+  - templates/deployment-configmap.yaml
+release:
+  name: my-release
+  namespace: my-namespace
+tests:
+  - it: uses default KSM selector when not specified
+    set:
+      cluster: my-cluster
+      licenseKey: us-whatever
+    asserts:
+      - matchRegex:
+          path: data["deployment-config.yaml"]
+          pattern: "regex: kube-state-metrics"
+        template: templates/deployment-configmap.yaml
+      - matchRegex:
+          path: data["deployment-config.yaml"]
+          pattern: "__meta_kubernetes_pod_label_app_kubernetes_io_name"
+        template: templates/deployment-configmap.yaml
```

Could we combine this into a more explicit test. As it's written, it's possible for the tests to pass if `regex: kube-state-metrics` and `__meta_kubernetes_pod_label_app_kubernetes_io_name` exist anywhere within the config. Instead, could we do the following:

```suggestion
      - template: templates/deployment-configmap.yaml
        matchRegex:
          path: data["deployment-config.yaml"]
          pattern: |
            ^\s*receivers:[\S\s]*prometheus/ksm:\s+config:\s+scrape_configs:\s+- job_name: kube-state-metrics\s+scrape_interval: 1m\s+kubernetes_sd_configs:\s+- role: pod\s+relabel_configs:\s+- action: keep\s+regex: kube-state-metrics\s+source_labels:\s+- __meta_kubernetes_pod_label_app_kubernetes_io_name\s+- action: replace\s+target_label: job_label\s+replacement: kube-state-metrics\s+
```

---

Good point!